src/common/Utility.js

```diff
@@ -420,6 +420,18 @@
                 }
                 return dest;
             }
+        },
+        stringify: function(obj) {
+            var cache = [];
```

Comment:
The code is written to only serialize the first instance of a given object, not the first instance of _any_ object.
As written it is over cautious as the same object could be referenced twice at the same level and that is not a circular reference - I will think about it some more (its not actually being called at the moment and I had been planning on only using it for console.log messages...) - I might rename it to logStringify